sonic_installer/main.py

```diff
@@ -208,6 +208,41 @@ def print_deprecation_warning(deprecated_cmd_or_subcmd, new_cmd_or_subcmd):
                 fg="red", err=True)
     click.secho("Please use '{}' instead".format(new_cmd_or_subcmd), fg="red", err=True)
 
+def update_sonic_environment(click, binary_image_version):
+    """Prepare sonic environment variable using incoming image template file. If incoming image template does not exist
+       use current image template file.
+    """
+    sonic_env_template_file = "/usr/share/sonic/templates/sonic-environment.j2"
+    sonic_version = re.sub("SONiC-OS-", '', binary_image_version)
+    new_image_squashfs_path = "/host/image-{0}/fs.squashfs".format(sonic_version)
+    new_image_mount = "/tmp/image-{0}-fs".format(sonic_version)
+
+    run_command("mkdir -p {0}".format(new_image_mount))
+    run_command("mount -t squashfs {0} {1}".format(new_image_squashfs_path, new_image_mount))
+
+    if os.path.exists(os.path.join(new_image_mount, sonic_env_template_file)):
+        sonic_env_template_file = os.path.join(new_image_mount, sonic_env_template_file)
+
+    if os.path.exists(sonic_env_template_file):
+        sonic_env = run_command_output(
+                "sonic-cfggen"
+                " -d"
+                " -y"
+                " /etc/sonic/sonic_version.yml"
```

Comment:
>/etc/sonic/sonic_version.yml [](start = 18, length = 28)

Could you use new image sonic_version.yml ?
Then you even don't need the function parameter `binary_image_version` ?

Reply:
I will remove the yml file, however the binary image version is required in order to know where the image is